packages/flame/lib/src/components/shape_component.dart

```diff
@@ -17,19 +23,29 @@ class ShapeComponent extends PositionComponent {
   @override
   final Anchor anchor = Anchor.center;
 
+  @override
+  set angle(double a) {
+    super.angle = a;
+    shape.angle = a;
+  }
+
   ShapeComponent(
     this.shape, {
     Paint? paint,
+    Vector2? scale,
     int? priority,
   })  : paint = paint ?? BasicPalette.white.paint(),
         super(
           position: shape.position,
           size: shape.size,
+          scale: scale,
           angle: shape.angle,
           anchor: Anchor.center,
           priority: priority,
         ) {
     shape.isCanvasPrepared = true;
+    position.addListener(() => shape.position.setFrom(position));
```

Comment:
We will need to clear those listeners when this component is removed right?

Reply:
Hmm, do we? Since this is listeners on itself, it shouldn't be a problem after it is removed right? And then if the component is re-added it will still have the listeners.

Reply:
Yeah, you are correct, I thought this was from the parent for some reason